public/inbox.php

```diff
@@ -175,24 +173,15 @@ function MarkAsUnread(msgID) {
 
             echo "</tbody></table>";
 
-            echo "<div class='float-right'>";
-
-            if ($offset > 0) {
-                echo "<a class='btn btn-link' href='/inbox.php?o=" . ($offset - $maxCount) . "&amp;u=$unreadOnly&amp;s=$outbox'>";
-                echo "&lt; Previous $maxCount";
-                echo "</a>";
-            }
-
-            if ($totalMsgs == $maxCount) {
-                echo "<a class='btn btn-link' href='/inbox.php?o=" . ($offset + $maxCount) . "&amp;u=$unreadOnly&amp;s=$outbox'>";
-                echo "Next $maxCount &gt;";
-                echo "</a>";
-            }
+            // Get message count and build paginator URL from current GET parameters
+            $messageCount = $unreadOnly ? $unreadMessageCount : $totalMessageCount;
+            unset($_GET['o']);
+            $urlPrefix = '/inbox.php?' . http_build_query($_GET) . '&o=';
 
+            echo "<div class='float-right'>";
+            RenderPaginator($messageCount, $displayCount, $offset, $urlPrefix);
```

Comment:
Noticed a small issue with the paginator when there are no items (unread only, no unread messages). Could you add this to layout.php?
```
     $pages = floor(($numItems + $perPage - 1) / $perPage);
+    if ($pages < 1) {
+        $pages = 1;
+    }
     for ($i = 1; $i <= $pages; $i++) {
```

Reply:
Okay, done. I have mixed feelings about whether we should show the paginator at all in the single page case, but... it's outside the scope of the PR.